pkg/apis/ingress/v1alpha1/status.go

```diff
@@ -0,0 +1,202 @@
+		if condition.Type == c.Type {
+			if condition.Status == c.Status &&
+				condition.Reason == c.Reason &&
+				condition.Message == c.Message {
+				return
+			}
+
+			found = true
+			conditions = append(conditions, *condition)
+		} else {
+			conditions = append(conditions, c)
+		}
+	}
+	if !found {
+		conditions = append(conditions, *condition)
+	}
+
+	ci.Status.Conditions = conditions
+}
+
+// SetStatusSyncCondition modifies the ClusterIngress object's status by setting
+// its SyncSuccessful condition in accordance with the provided error value.
+func (ci *ClusterIngress) SetStatusSyncCondition(err error) {
+	condition := &operatorapi.OperatorCondition{
+		Type:               operatorapi.OperatorStatusTypeSyncSuccessful,
+		Status:             operatorapi.ConditionTrue,
+	}
+
+	if err != nil {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "SyncFailed"
+		condition.Message = err.Error()
+	}
+
+	ci.SetStatusCondition(condition)
+}
+
+// SetStatusAvailableCondition modifies the ClusterIngress object's status by
+// setting its Available condition to the specified status and message.
+func (ci *ClusterIngress) SetStatusAvailableCondition(available bool, message string) {
+	condition := &operatorapi.OperatorCondition{
+		Type:               operatorapi.OperatorStatusTypeAvailable,
+		Status:             operatorapi.ConditionTrue,
+	}
+
+	if !available {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "ServiceNotReady"
+		condition.Message = message
+	}
+
+	ci.SetStatusCondition(condition)
+}
+
+// clusterIngressStatusEqual returns true iff the status conditions (ignoring
+// LastTransitionTime) and ingresses of the provided ClusterIngress objects are
+// equal.
+func clusterIngressStatusEqual(oldCi, ci *ClusterIngress) bool {
+	numConditions := len(ci.Status.Conditions)
+	if numConditions != len(oldCi.Status.Conditions) {
+		return false
+	}
+	for i := 0; i < numConditions; i++ {
+		conditionA := oldCi.Status.Conditions[i]
+		conditionB := ci.Status.Conditions[i]
+
+		// Compare every field except LastTransitionTime.
+		if conditionA.Type != conditionB.Type ||
+			conditionA.Status != conditionB.Status ||
+			conditionA.Reason != conditionB.Reason ||
+			conditionA.Message != conditionB.Reason {
+			return false
+		}
+	}
+
+	numIngresses := len(ci.Status.LoadBalancer.Ingress)
+	if numIngresses != len(oldCi.Status.LoadBalancer.Ingress) {
+		return false
+	}
+	for i := 0; i < numIngresses; i++ {
+		ingressA := oldCi.Status.LoadBalancer.Ingress[i]
+		ingressB := ci.Status.LoadBalancer.Ingress[i]
+
+		if ingressA != ingressB {
+			return false
+		}
+	}
+
+	return true
+}
+
+// UpdateStatus updates the receiver ClusterIngress object's status if it
+// differs from the provided ClusterIngress object's status.
+func (ci *ClusterIngress) UpdateStatus(oldCi *ClusterIngress) {
+	if clusterIngressStatusEqual(oldCi, ci) {
+		return
+	}
+
+	err := sdk.Update(ci)
+	for tries := 3; tries > 0; tries-- {
```

Comment:
Why is the loop needed?
the err variable is updated outside the loop or am i missing something?

Reply:
Whoops! `err := sdk.Update(ci)` needs to be inside the loop.  Thanks!

Reply:
I deleted the loop.  Now `UpdateStatus` returns an error value, and callers return any error from `UpdateStatus` to the handler so that the event gets reprocessed.